core/domain/state_domain.py

```diff
@@ -1790,29 +1790,20 @@ def update_interaction_hints(self, hints_list):
         self._update_content_ids_in_assets(
             old_content_id_list, new_content_id_list)
 
-    def update_interaction_solution(self, solution_dict):
+    def update_interaction_solution(self, solution):
         """Update the solution of interaction.
 
         Args:
-            solution_dict: dict or None. The dict representation of
-                Solution object.
-
-        Raises:
-            Exception: 'solution_dict' is not a dict.
+            solution: Solution or None. The Solution object.
         """
         old_content_id_list = []
         new_content_id_list = []
         if self.interaction.solution:
             old_content_id_list.append(
                 self.interaction.solution.explanation.content_id)
 
-        if solution_dict is not None:
-            if not isinstance(solution_dict, dict):
-                raise Exception(
-                    'Expected solution to be a dict, received %s'
-                    % solution_dict)
-            self.interaction.solution = Solution.from_dict(
-                self.interaction.id, solution_dict)
+        if solution is not None:
+            self.interaction.solution = solution
```

Comment:
Do we still need this line inside the `if-else` statement?

Reply:
Moved the assignment outside the if-else.